src/http.rs

```diff
@@ -168,41 +194,29 @@ mod tests {
         )
         .unwrap();
 
-        let request_builder = http_client.client.post("http://example.com");
```

Comment:
it was easy to test post or get but it didn't test the whole code in the http client. Now that we use get_json or post_json we have much better coverage but at a price of testing async code